source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -122,16 +107,40 @@ void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::strin
   callbacks_ = nullptr;
 }
 
-void GrpcClientImpl::toAuthzResponseHeader(
-    ResponsePtr& response,
-    const Protobuf::RepeatedPtrField<envoy::config::core::v3::HeaderValueOption>& headers) {
+void GrpcClientImpl::copyHeaderMutationsIntoResponse(
+    ResponsePtr& response, const RepeatedHeaderValueOption& headers,
+    const RepeatedHeaderValueOption& response_headers_to_add,
+    const Protobuf::RepeatedPtrField<std::string>& headers_to_remove) {
   for (const auto& header : headers) {
-    if (header.append().value()) {
-      response->headers_to_append.emplace_back(Http::LowerCaseString(header.header().key()),
-                                               header.header().value());
-    } else {
-      response->headers_to_set.emplace_back(Http::LowerCaseString(header.header().key()),
-                                            header.header().value());
+    if (Http::HeaderUtility::headerNameIsValid(header.header().key()) &&
+        Http::HeaderUtility::headerValueIsValid(header.header().value())) {
```

Comment:
I think silently dropping requested mutations is a mistake. I'd rather fail the entire client request in this case. 

A runtime guard on the behavior change (or config) is needed.

Reply:
Failing on invalid mutations is also possible, but the side stream has already indicated that the client request should be accepted. Perhaps it should be configurable?

Reply:
I agree with your reasoning!

Reply:
Reject the whole request is good, but I think we should make sure such rejection caused by invalid mutation is different from existing rejection caused by authorization failure. (e.g., the response code and responses body in localReply should be different)

Reply:
Agreed; the response code should be in the 5xx range, because this is a server-side bug in the auth service.

Reply:
Done.